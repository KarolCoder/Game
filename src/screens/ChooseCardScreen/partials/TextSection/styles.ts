import styled from 'styled-components/native';

export const CategoryBox = styled.View`
  justify-content: center;
  align-items: center;
`;
