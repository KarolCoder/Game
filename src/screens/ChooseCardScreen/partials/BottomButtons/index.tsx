import {Spacer} from '@/components';
import React from 'react';
import {Row} from '../../styles';
import {BottomButton} from './styles';

interface BottomButtonsProps {
  isDisabled?: boolean;
  onLeftPress: () => void;
  onRightPress: () => void;
}

export const BottomButtons: React.FC<BottomButtonsProps> = ({
  isDisabled,
  onLeftPress,
  onRightPress,
}) => {
  return (
    <Row>
      <BottomButton
        mode="contained"
        disabled={isDisabled}
        onPress={onLeftPress}>
        Reset cards
      </BottomButton>
      <Spacer width={24} />
      <BottomButton
        mode="contained"
        disabled={isDisabled}
        onPress={onRightPress}>
        Go to start
      </BottomButton>
    </Row>
  );
};
