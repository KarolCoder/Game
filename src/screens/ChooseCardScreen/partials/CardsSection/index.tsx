import {Spacer} from '@/components';
import React from 'react';
import {Title, useTheme} from 'react-native-paper';
import {Row} from '../../styles';
import {Card, SelectTitle} from './styles';

interface CardsSectionProps {
  setSelectedCard: React.Dispatch<React.SetStateAction<number | undefined>>;
  refreshData: () => void;
  getNextPage: () => Promise<void>;
  firstValueDisplayed?: string;
  secondValueDisplayed?: string;
  resourcesFromData?: {
    firstResource: string | string[] | undefined;
    secondResource: string | string[] | undefined;
  };
  nextPage?: string;
  status?: string;
}

export const CardsSection: React.FC<CardsSectionProps> = ({
  resourcesFromData,
  setSelectedCard,
  refreshData,
  getNextPage,
  firstValueDisplayed,
  secondValueDisplayed,
  nextPage,
  status,
}) => {
  const {colors} = useTheme();
  return (
    <>
      <SelectTitle textColor={colors.primary}>Choose card</SelectTitle>
      <Row>
        <Card
          disabled={
            (!!resourcesFromData?.firstResource &&
              !!resourcesFromData?.secondResource) ||
            status === 'fetching'
          }
          onPress={() => {
            setSelectedCard(0);
            if (!nextPage) {
              refreshData();
            } else {
              getNextPage();
            }
          }}>
          <Title>{firstValueDisplayed ? firstValueDisplayed : '?'}</Title>
        </Card>
        <Spacer width={24} />
        <Card
          disabled={
            (!!resourcesFromData?.firstResource &&
              !!resourcesFromData?.secondResource) ||
            status === 'fetching'
          }
          onPress={() => {
            setSelectedCard(1);
            if (!nextPage) {
              refreshData();
            } else {
              getNextPage();
            }
          }}>
          <Title>{secondValueDisplayed ? secondValueDisplayed : '?'}</Title>
        </Card>
      </Row>
    </>
  );
};
