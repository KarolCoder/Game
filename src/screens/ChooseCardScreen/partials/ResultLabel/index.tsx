import {BigText} from '../../styles';
import React from 'react';
import {useTheme} from 'react-native-paper';
import {GameResult} from '@/utils/types';

interface ResultLabelProps {
  result: GameResult;
}

export const ResultLabel: React.FC<ResultLabelProps> = ({result}) => {
  const {colors} = useTheme();

  return result === 'Winner' ? (
    <BigText textColor={colors.primary}>Congratualtions, you won!!!</BigText>
  ) : result === 'Loser' ? (
    <BigText textColor={colors.primary}>You lose, loser :D</BigText>
  ) : result === 'Tie' ? (
    <BigText textColor={colors.primary}>Tie, try again</BigText>
  ) : !result ? (
    <BigText textColor={colors.primary}>
      Couldn't resolve winner please go to start
    </BigText>
  ) : null;
};
