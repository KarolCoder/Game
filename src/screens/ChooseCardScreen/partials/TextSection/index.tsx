import React from 'react';
import {useTheme} from 'react-native-paper';
import {BigText, Row, SmallText} from '../../styles';
import {CategoryBox} from './styles';

interface TextSectionProps {
  leftTopText?: string;
  rightTopText?: string;
  bottomLeftText?: string | number;
  bottomRightText?: string | number;
}

export const TextSection: React.FC<TextSectionProps> = ({
  leftTopText = '',
  rightTopText = '',
  bottomLeftText = '',
  bottomRightText = '',
}) => {
  const {colors} = useTheme();
  return (
    <Row>
      <CategoryBox>
        <SmallText textColor={colors.primary}>{leftTopText}</SmallText>
        <BigText textColor={colors.primary}>{bottomLeftText}</BigText>
      </CategoryBox>
      <CategoryBox>
        <SmallText textColor={colors.primary}>{rightTopText}</SmallText>
        <BigText textColor={colors.primary}>{bottomRightText}</BigText>
      </CategoryBox>
    </Row>
  );
};
