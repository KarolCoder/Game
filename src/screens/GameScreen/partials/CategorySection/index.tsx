import {Spacer} from '@/components';
import {GameCategory} from '@/utils/types';
import React from 'react';
import {useTheme} from 'react-native-paper';
import {Row, SelectTitle} from '../../styles';
import {OptionButton} from './styles';

interface CategorySectionProps {
  activeCategory?: GameCategory;
  setActiveCategory: (category?: GameCategory | undefined) => void;
}

export const CategorySection: React.FC<CategorySectionProps> = ({
  setActiveCategory,
  activeCategory,
}) => {
  const {colors} = useTheme();
  return (
    <>
      <SelectTitle textColor={colors.primary}>Choose category</SelectTitle>
      <Row>
        <OptionButton
          testID="option button 1"
          isActive={activeCategory === 'People'}
          mode="contained"
          onPress={() => setActiveCategory('People')}>
          People
        </OptionButton>
        <Spacer width={24} />
        <OptionButton
          testID="option button 2"
          isActive={activeCategory === 'Starships'}
          mode="contained"
          onPress={() => setActiveCategory('Starships')}>
          Starships
        </OptionButton>
      </Row>
    </>
  );
};
