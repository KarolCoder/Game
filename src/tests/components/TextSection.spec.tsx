import {TextSection} from '@/screens/ChooseCardScreen/partials/TextSection';
import {render} from '@testing-library/react-native';
import React from 'react';

import renderer from 'react-test-renderer';

it('text sections renders correctly', async () => {
  const component = (
    <TextSection
      bottomLeftText={'bottomLeft'}
      bottomRightText={'bottomRight'}
      leftTopText="leftTop"
      rightTopText="rightTop"
    />
  );
  const result = renderer.create(component).toJSON();
  const {getByText} = render(component);
  expect(result).toMatchSnapshot();
  getByText('leftTop');
  getByText('bottomRight');
  getByText('rightTop');
  getByText('bottomLeft');
});
