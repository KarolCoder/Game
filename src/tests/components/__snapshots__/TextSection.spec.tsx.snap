// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`text sections renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "justifyContent": "space-evenly",
        "paddingBottom": 24,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 24,
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Array [
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontFamily": "System",
              "fontWeight": "400",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "fontSize": 14,
                "letterSpacing": 0.25,
                "lineHeight": 20,
                "marginVertical": 2,
              },
              Array [
                Object {
                  "color": "#6200ee",
                },
              ],
            ],
          ],
        ]
      }
      textColor="#6200ee"
    >
      leftTop
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Array [
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontFamily": "System",
              "fontWeight": "500",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "fontSize": 20,
                "letterSpacing": 0.15,
                "lineHeight": 30,
                "marginVertical": 2,
              },
              Array [
                Object {
                  "color": "#6200ee",
                },
              ],
            ],
          ],
        ]
      }
      textColor="#6200ee"
    >
      bottomLeft
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Array [
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontFamily": "System",
              "fontWeight": "400",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "fontSize": 14,
                "letterSpacing": 0.25,
                "lineHeight": 20,
                "marginVertical": 2,
              },
              Array [
                Object {
                  "color": "#6200ee",
                },
              ],
            ],
          ],
        ]
      }
      textColor="#6200ee"
    >
      rightTop
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Array [
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontFamily": "System",
              "fontWeight": "500",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "fontSize": 20,
                "letterSpacing": 0.15,
                "lineHeight": 30,
                "marginVertical": 2,
              },
              Array [
                Object {
                  "color": "#6200ee",
                },
              ],
            ],
          ],
        ]
      }
      textColor="#6200ee"
    >
      bottomRight
    </Text>
  </View>
</View>
`;
